tmt/steps/execute/__init__.py

```diff
@@ -532,7 +532,7 @@ def prepare_scripts(self, guest: "tmt.steps.provision.Guest") -> None:
         """ Prepare additional scripts for testing """
         # Install all scripts on guest
         for script in self.scripts:
-            source = SCRIPTS_SRC_DIR / script.path.name
+            source = Path(str(SCRIPTS_SRC_DIR / script.path.name))
```

Comment:
I will need some help to understand the various effects of these changes, bear with me, please.

IIUIC, `SCRIPTS_SRC_DIR` is no longer `Path` instance, but `MultiplexedPath` ("If the final path is a file, go ahead and convert it to a regular ``Path``, otherwise keep it as a traversable in order to properly support MultiplexedPaths." - and `SCRIPTS_SRC_DIR` is definitely not pointing to a file. What effect does this multiplexed property have in loops like this one? tmt tries to construct a path from this now `MultiplexedPath` and a script file name, do I understand it correctly there might be multiple actual directories "hidden" behind the `SCRIPTS_SRC_DIR`? Shouldn't we somehow iterate over them to find the script? Maybe it's not present in all of those directories. Would it also make sense to support the addition of the actual scripts? Because without it, having an extra path lets me override known scripts, but I can't add new ones.

Reply:
Here's the beauty of how `MultiplexedPath` works, it does all of that for you. I'm on the phone now so I can't make the directory tree, but let's consider we have a `mp = MultiplexedPath(A, B)`.
- if `A / c` and `B / c` both exist, then `mp / c` is a MultiplexedPath
- if only one of `A / c` or `B / c` exists, `mp / c` is converted to the regular Path where it exists
- if neither `A / c` nor `B / c` exists, `mp / c` is converted randomly to a regular Path to one of them (first one)

Now if `mp / c` points to a file than it will always be a `Path`, because MultiplexedPath errors if components are not directories. If there are multiple files, than it is illdefined by the definition of Traversable, but it failsafes to the first file.

Design wise we must ask plugin designers to play nice and don't create filestructures that can clash, usually by appropriately prefixing their classes, templates, schemas, etc.

Reply:
Oh, need to bounce something passed you @happz. I can basically change this part to:
```python
            source = SCRIPTS_SRC_DIR / script.path.name
            assert isinstance(source, pathlib.Path)
```
Or better yet add a proper check that it is a file and not a directory. But the issue is that we would need to convert `pathlib.Path` to `tmt.utils.Path` (`tmt._compat.pathlib.Path`). It's quite tricky because it can't be done at the `SCRIPTS_SRC_DIR` because that part can be a `MultiplexedPath`, and it has to be done at the endpoint. What interface do you think would be good to translate a `Traversable`/`pathlib.Path` to `tmt.utils.Path`?

What is puzzling me is the `relative_to` and in what context is it needed to override the original behavior there?